Some fixes and new features

norowachi opened the pull request:
* changed `econ balance` to `econ bal`
* fixed the typos (and ctrl+c ctrl+v traces) in register
* fixed some starboard stuff and added it to the settings command (also made it a command to set the starboard channel)
* Made the warn system (best warning system in the world!!1! or so)
* also i fixed some things in the settings command and added starboard to `settings view`
* obv had to add warn interfaces and types so yea
* added the `help query: xxx` command to search in command (obv?)
* - changed channel to channelId (in the starboard schema)
* - replaced checkPermsBoth with CheckPerms in the settings command group ( c: )
* Added tags (and only message command tags, starts with `t.`)
* Made environment variables accessable from config.env
* - ~~planning to make it support multiple words tag (ex: `t.hello world` instead of `t.helloworld`) and also adding tags limit (20-35 didnt decide yet)~~ made it only one word
* Made dev commands only usable by message commands (mention or prefix, it also works with spaces, ex: `mc! help`)
* Changed BaseCommand to BaseInt and created BaseMsg (for message commands)
* Made a register function for dev commands and another property on client (also created `client.prefix`)
* Edited Clean function so that it doesn't add `\` before the styling stuff only the "`" (which i forgot its name)
* Updated Eval so that it can support async code (updated again, it now supports `return`, tho you'll have to use it if you want to "get" a value back...)
* Made levelxp's user ref to a new user schema (`{id: string, tag: string, avatarHash: string}`) and created `id` to carry the user's id, to be able to link the schema(s) easier.

Note: avatarHash is b0226aa2a8ad74b6f71e2f2a1208edb8 in
https://cdn.discordapp.com/avatars/534783899331461123/b0226aa2a8ad74b6f71e2f2a1208edb8.webp
can also be a number (from 1 to 5) if user has no avatar (aka default avatar)

norowachi committed:
Added starboard, fixed welcome

Added starboard (i'm not too sure how fine is it)
Fixed welcome
and made some small updates: changed _ads.onCooldown to make more sense
Added "Event" to each event file, also exported  the econ interfaces and imported them to the econ function file and finally fixed spelling in report command.
oh almost forgot, i fixed ani char so it doesn't give an error when the character is not found.
almost forgot again, added GUILD_MESSAGE_REACTIONS intent, allowed user mentions only, and enabled partial CHANNEL, MESSAGE, and REACTION (all that when defining client in src/index.ts)

norowachi committed:
Merge branch 'OpenianDevelopment:new' into new

norowachi committed:
added warn schema, interfaces and functions

norowachi committed:
Added "command" command

norowachi committed:
Some fixed and added warn commands

* changed `econ balance` to `econ bal`
* fixed the typos (and ctrl+c ctrl+v traces) in register
* fixed some starboard stuff and added it to the settings command (also made it a command to set the starboard channel)
* Made the warn system (best warning system in the world!!1! or so)
* also i fixed some things in the settings command and added starboard to `settings view`
* obv had to add warn interfaces and types so yea
* added the `help query: xxx` command to search in command (obv?)
** changed channel to channelId (in the starboard schema)
** replaced checkPermsBoth with CheckPerms in the settings command group ( c: )

norowachi committed:
added tags

norowachi committed:
Update eval.ts

norowachi committed:
Update eval.ts

norowachi committed:
Update eval.ts

norowachi committed:
Update clean function

i mean, styles wont affect us cuz we're already using them in `this format` so yeah, having those `\_id: "000";` looks bad, so now it only works on ` and @

norowachi committed:
Dev Message Commands

* Finished tags
* Made dev commands only usable by message commands (mention or prefix, it also works with spaces, ex: `mc! help`)
* Changed BaseCommand to BaseInt and created BaseMsg (for message commands)
* Made a register function for dev commands and another property on client (also created `client.prefix`)
* There is a chance that there are some changes/edits/updates (whatever) that i didnt right here

p.s. code still need some more reviews from a 2nd party (iyp)

norowachi committed:
Update register.ts

Made register support command line arguments, and if there are no args it'll ask `Do you to [create] new commands or [delete] current ones? `
command-line ex:
* `yarn register create` or `yarn register c` to create.
* `yarn register delete` ~~or  `yarn register deez nuts`~~ to delete
(ig you got how it works)
prompt ex:
```bash
$ yarn register

> Do you to [create] new commands or [delete] current ones?
#where you can answer with [d]elete or [c]reate (to do their respective jobs obviously)
```

july12123 reviewed (approved):
as far as i could tell there wasn't anything i could see as bad but i don't know if i missed something

norowachi committed:
Update register.ts

ehm.. fixed my english

norowachi committed:
ref levelxp's user to another schema

* Made levelxp's user ref to a new user schema (`{id: string, tag: string, avatarHash: string}`) and created `id` to carry the user's id, to be able to link the schema(s) easier.
* Fixed my english (forgot "want" in the question prompt, register file...)

p.s. avatarHash is b0226aa2a8ad74b6f71e2f2a1208edb8 in
https://cdn.discordapp.com/avatars/534783899331461123/b0226aa2a8ad74b6f71e2f2a1208edb8.webp
can also be a number (from 1 to 5) if user has no avatar (aka default avatar)
